apps/storefront/src/shared/service/b2b/graphql/global.ts

```diff
@@ -171,6 +171,47 @@ const storefrontConfig = () => `{
 	}
 }`;
 
+const storefrontConfigWithCompanyHierarchy = () => `
+query storefrontConfigWithCompanyHierarchy($includeCompanyHierarchy: Boolean!) {
+	storefrontConfig(
+		storeHash: "${storeHash}"
+	) {
+		config{
+			accountSettings,
+			addressBook,
+			buyAgain,
+			dashboard,
+			invoice{
+				enabledStatus,
+				value,
+			},
+			messages,
+			orders,
+			quickOrderPad,
+			quotes,
+			recentlyViewed,
+			returns,
+			shoppingLists,
+			tradeProfessionalApplication,
+			userManagement,
+			wishLists,
+		}
+		configId,
+	}
+	companySubsidiaries @include(if: $includeCompanyHierarchy) {
+		companyId
+		companyName
+		parentCompanyId
+		parentCompanyName
+		channelFlag
+	}
+	userMasqueradingCompany @include(if: $includeCompanyHierarchy) {
+		companyId
+		companyName
+		bcId
+	}
+}`;
+
```

Comment:
updated the combined query to conditionally request `companySubsidiaries` and `userMasqueradingCompany` only if user has permissions to company hierarchy info